crankshaft-engine/src/lib.rs

```diff
@@ -20,14 +26,27 @@ use crate::service::runner::TaskHandle;
 pub struct Engine {
     /// The task runner(s).
     runners: IndexMap<String, Runner>,
+    /// The event sender, if monitoring is enabled.
+    event_sender: Option<broadcast::Sender<Event>>,
+    /// The monitoring server task handle
+    monitoring_handle: Option<JoinHandle<Result<(), tonic::transport::Error>>>,
 }
 
 impl Engine {
     /// Adds a [`Backend`] to the engine.
     pub async fn with(mut self, config: Config) -> Result<Self> {
-        let (name, kind, max_tasks, defaults) = config.into_parts();
-        let runner = Runner::initialize(kind, max_tasks, defaults).await?;
+        let (name, kind, max_tasks, defaults, monitored) = config.into_parts();
+        let runner = Runner::initialize(kind, max_tasks, defaults, monitored).await?;
         self.runners.insert(name, runner);
+
+        // Start the monitoring server if any runner is monitored
+        if monitored && self.event_sender.is_none() {
+            let socketaddr: SocketAddr = "127.0.0.1:8080".parse()?;
```

Comment:
We can address this in a follow-up PR, but the listen address should be part of configuration.